corehq/apps/userreports/models.py

```diff
@@ -753,6 +753,8 @@ class StaticDataSourceConfiguration(JsonObject):
     domains = ListProperty(required=True)
     server_environment = ListProperty(required=True)
     config = DictProperty()
+    # List of engine_ids to be mirrored to by environment
+    mirrored_engine_ids = DictProperty()
```

Comment:
i think this should be a list property in `DataSourceConfigration` as well. Then in [_get_datasource_config](https://github.com/dimagi/commcare-hq/blob/ecd8e4862419168cf4cd9a7d5af162ca72acde24/corehq/apps/userreports/models.py#L819) you can convert this dict into a list based on the server env

Reply:
I thought, we only wanted this on static data sources and not the dynamic ones, so I had put it just here.

Reply:
@emord  re. `SERVER_ENVIRONEMTN` see https://github.com/dimagi/commcare-hq/pull/24046#issuecomment-484353973

Reply:
All `StaticDataSourceConfiguration`'s end up being converted to a `DataSourceConfiguration` (one for each domain in the list of domains). See `_get_datasource_config`.

Re the `SERVER_ENVIRONMENT` question, I don't have a strong opinion other than to say that specifying it here adds a second place in the config where it's referenced.